src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -1785,6 +1825,47 @@ public DagState(ImmutableArray<StateForCase> cases, ImmutableDictionary<BoundDag
             // After the entire graph of DagState objects is complete, we translate each into its Dag node.
             public BoundDecisionDagNode? Dag;
 
+            private DagState()
+            {
+            }
+
+            public void ClearAndFree()
```

Comment:
nit: consider moving `Create` above `ClearAndFree` and rename `Create` to `GetInstance`

Reply:
Done.